networkit/cpp/graph/Graph.cpp

```diff
@@ -179,6 +179,19 @@ void Graph::preallocateUndirected(node u, size_t size) {
 	}
 }
 
+void Graph::preallocateDirected(node u, size_t outSize, size_t inSize) {
```

Comment:
Assert that (also for `preallocateUndirected`)
- the graph is directed
- `u` is in range

Probably we do not want to assert that `outSize` and `inSize` are within range, as a user might know that the graph will be growing in future?